rfc/src/rfcs/0005-should-fail-attr.md

```diff
@@ -0,0 +1,197 @@
+
+We don't expect these changes to require new dependencies.
+Besides, we don't expect these changes to be updated unless we decide to extend the attribute with further fields (see [Future possibilities](#future-possibilities) for more details).
+
+## Rationale and alternatives
+
+This proposal would enable users to exercise negative verification with a relatively simple mechanism.
+Moreover, it could be used in the Kani regression and remove the special-cased code to handle negative harnesses in `compiletest`.
+
+Not adding such a mechanism could impact Kani's usability by limiting the harnesses that users can write.
+
+### Alternative #1: Name
+
+Our first choice is made in the name of the attribute.
+It should be noted that Rust has a similarly named attribute for unit tests: [the `#[should_panic]` attribute](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#testing-panics).
+At first, one may think that `#[kani::should_fail]` and `#[should_panic]` have the same semantics.
+However, if we keep thinking about this, we'll realize that:
+ 1. A panic in Rust should always result in a failure in Kani, but
+ 2. A failure in Kani doesn't always result in a panic in Rust.
+
+Our running example shows an instance of (2): an overflow error doesn't result in a panic in Rust.
+So it's better to avoid `panic` in the name.
+
+We have also considered two alternatives for the expectation: `should` and `expect`.
+To be honest, we avoid `expect` altogether for two reasons:
+ - We may consider a more granular approach to indicate expectations regarding individual checks and cover statements in the future. One possible name for the attribute is `#[kani::expect]`.
+ - We heavily use this word for testing in Kani: there is an `expected` mode, which works with `*.expected` files. Other modes also use such files.
+
+### Alternative #2: Granularity
+
+As mentioned earlier, users may want to express more granular expectations for their harnesses.
+For example, a user may want to specify that a given check results in a failure.
+
+Again, there is a relation between this and the [the `#[should_panic]` attribute](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#testing-panics) for Rust tests.
+More specifically, the `#[should_panic]` attribute may receive an argument `expected` which allows users to specify the expected panic string:
+
+```rust
+    #[test]
+    #[should_panic(expected = "Divide result is zero")]
+    fn test_specific_panic() {
+        divide_non_zero_result(1, 10);
+    }
+```
+
+In principle, it wouldn't be a problem to extend this proposal to include the `expected` argument.
+The implementation could compare the `expected` string against property descriptions.
+
+However, there could be some problems later with this approach:
+ * What if users want to specify multiple `expected` strings?
```

Comment:
could users have multiple annotations?

Reply:
Another good point, I'll add that to this list.

Reply:
The RFC now promotes a discussion around multiple panics, and suggests the panic expectation be limited to one. It isn't clear to me that we want to allow multiple panics with this attribute, but there's an open question related to this aspect.